addons/binding/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/device/PJLinkDevice.java

```diff
@@ -0,0 +1,321 @@
+import org.openhab.binding.pjlinkdevice.internal.device.command.ResponseException;
+import org.openhab.binding.pjlinkdevice.internal.device.command.authentication.AuthenticationCommand;
+import org.openhab.binding.pjlinkdevice.internal.device.command.errorstatus.ErrorStatusQueryCommand;
+import org.openhab.binding.pjlinkdevice.internal.device.command.errorstatus.ErrorStatusQueryResponse.ErrorStatusDevicePart;
+import org.openhab.binding.pjlinkdevice.internal.device.command.errorstatus.ErrorStatusQueryResponse.ErrorStatusQueryResponseState;
+import org.openhab.binding.pjlinkdevice.internal.device.command.identification.IdentificationCommand;
+import org.openhab.binding.pjlinkdevice.internal.device.command.input.Input;
+import org.openhab.binding.pjlinkdevice.internal.device.command.input.InputInstructionCommand;
+import org.openhab.binding.pjlinkdevice.internal.device.command.input.InputListQueryCommand;
+import org.openhab.binding.pjlinkdevice.internal.device.command.input.InputQueryCommand;
+import org.openhab.binding.pjlinkdevice.internal.device.command.input.InputQueryResponse;
+import org.openhab.binding.pjlinkdevice.internal.device.command.mute.MuteInstructionCommand;
+import org.openhab.binding.pjlinkdevice.internal.device.command.mute.MuteInstructionCommand.MuteInstructionChannel;
+import org.openhab.binding.pjlinkdevice.internal.device.command.mute.MuteInstructionCommand.MuteInstructionState;
+import org.openhab.binding.pjlinkdevice.internal.device.command.mute.MuteQueryCommand;
+import org.openhab.binding.pjlinkdevice.internal.device.command.mute.MuteQueryResponse.MuteQueryResponseValue;
+import org.openhab.binding.pjlinkdevice.internal.device.command.power.PowerInstructionCommand;
+import org.openhab.binding.pjlinkdevice.internal.device.command.power.PowerQueryCommand;
+import org.openhab.binding.pjlinkdevice.internal.device.command.power.PowerQueryResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Nils Schnabel - Initial contribution
+ */
+public class PJLinkDevice {
+    private static final int TIMEOUT = 30000;
+    protected int tcpPort;
+    protected InetAddress ipAddress;
+    protected String adminPassword;
+    protected boolean authenticationRequired;
+    protected BufferedReader reader;
+    protected Socket socket;
+    protected int timeout = TIMEOUT;
+    private Logger logger;
+    private String prefixForNextCommand;
+    private Instant socketCreatedOn;
+
+    public PJLinkDevice(int tcpPort, InetAddress ipAddress, String adminPassword, int timeout) {
+        super();
+
+        this.tcpPort = tcpPort;
+        this.ipAddress = ipAddress;
+        this.adminPassword = adminPassword;
+        this.timeout = timeout;
+    }
+
+    public PJLinkDevice(int tcpPort, InetAddress ipAddress, String adminPassword) {
+        this(tcpPort, ipAddress, adminPassword, TIMEOUT);
+    }
+
+    protected Logger getLogger() {
```

Comment:
Why not have a field:`private final Logger logger = LoggerFactory.getLogger(PJLinkDevice.class);`

Reply:
I think that was a remained from when I included the device address in the logger scope. Now it was just useless and is replaced by the field.